### PR TITLE
issue-86 - Export image of automaton

### LINKED ISSUE #83: Export image of automaton
Users should be able to export an image of their automaton, that could be submitted for an assignment for example.

### PR DESCRIPTION
**Pull request by nilcinova**
Issue: https://github.com/AutomatonBuilderGUI/AutomatonBuilderGUI/issues/83 

Export of the state diagram

The new _getDiagramBounds_ function calculates the bounds of the diagram on the canvas by iterating through nodes and transitions to determine their minimum and maximum positions and also adding padding for aesthetics. The _exportAutomatonToImage_ function temporarily resets the stage's zoom and position, adds a white background behind the diagram, and exports the defined bounds as a PNG image. Then puts the canvas into the original state after the export. These functions make sure that the exported imagecontains the exact diagram with a clean background, regardless of zoom.

Malcolm edit: Closes #83

### CODE REVIEW

File: src/StateManager.ts [hunk 1]
```diff
@@ -270,11 +270,114 @@ export default class StateManager {
...
+        StateManager._nodeWrappers.forEach((node) => {
+            const { x, y } = node.nodeGroup.position();
+            const radius = NodeWrapper.NodeRadius;
+            minX = Math.min(minX, x - radius);
+            minY = Math.min(minY, y - radius);
+            maxX = Math.max(maxX, x + radius);
+            maxY = Math.max(maxY, y + radius);
+        });
+
+        StateManager._transitionWrappers.forEach((transition) => {
```
Comment (Meorge) [inline src/StateManager.ts:294]: Is it necessary to check the source and destination nodes for the transition wrappers? I think all nodes that could be the source or destination for a transition would already have been iterated over within the node wrapper array. (If there is a reason to do this iteration over the transitions, though, we can certainly keep it!)

**Review by Meorge — COMMENTED** [review]
Looks great overall! There's just one thing in `getDiagramBounds` that I thought might be unnecessary

File: src/StateManager.ts (hunk 1 above)
Comment (nilcinova) [reply src/StateManager.ts:294]: Ou yes, you are right. xD Those are not needed, I am gonna remove them. :)

**Comment by Meorge** [comment]
Looks great, thank you again! 🥳

**Pull request merged**